### ISSUE TITLE
Puma crashes when col_sep=>nil

### ISSUE THREAD

**Issue by Shweit** [tilo/smarter_csv#216]
I noticed that the web server crashes as soon as you use the value nil for the 'col_sep' within the options. Wouldn't there be a simple way within the gem to start a simple query if a passed value is nil?

That would be fatal if something like that would happen on an operational website.

Here is some more information about the crash and some code:
```ruby
@csv = CsvFile.find(params[:id])
separator = if params[:separator] == "SPACE" then ' ' else end

if params[:skipLine]
  options = { skip_lines: 1, col_sep: separator, row_sep: :auto, file_encoding: @csv.encoding, remove_empty_values: false }
  session[:skipLine] = true
  session[:separator] = params[:separator]
end

@data = SmarterCSV.process(@csv.filepath, options)
```
This code leads to a complete crash of the WebServer.

Here is the Stacktrace:
```
-- Ruby level backtrace information ----------------------------------------
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/puma-3.12.6/lib/puma/thread_pool.rb:135:in `block in spawn_thread'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/puma-3.12.6/lib/puma/server.rb:334:in `block in run'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/puma-3.12.6/lib/puma/server.rb:476:in `process_client'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/puma-3.12.6/lib/puma/server.rb:706:in `handle_request'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/puma-3.12.6/lib/puma/configuration.rb:227:in `call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/railties-7.0.4.1/lib/rails/engine.rb:530:in `call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rack-proxy-0.7.6/lib/rack/proxy.rb:87:in `call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/webpacker-4.3.0/lib/webpacker/dev_server_proxy.rb:23:in `perform_request'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/action_dispatch/middleware/host_authorization.rb:137:in `call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rack-2.2.6.2/lib/rack/sendfile.rb:110:in `call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/action_dispatch/middleware/static.rb:23:in `call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/action_dispatch/middleware/executor.rb:14:in `call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/action_dispatch/middleware/server_timing.rb:60:in `call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/action_dispatch/middleware/server_timing.rb:26:in `collect_events'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/action_dispatch/middleware/server_timing.rb:61:in `block in call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/activesupport-7.0.4.1/lib/active_support/cache/strategy/local_cache_middleware.rb:29:in `call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rack-2.2.6.2/lib/rack/runtime.rb:22:in `call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rack-2.2.6.2/lib/rack/method_override.rb:24:in `call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/action_dispatch/middleware/request_id.rb:26:in `call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/action_dispatch/middleware/remote_ip.rb:93:in `call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/sprockets-rails-3.4.2/lib/sprockets/rails/quiet_assets.rb:13:in `call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/railties-7.0.4.1/lib/rails/rack/logger.rb:25:in `call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/activesupport-7.0.4.1/lib/active_support/tagged_logging.rb:99:in `tagged'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/activesupport-7.0.4.1/lib/active_support/tagged_logging.rb:37:in `tagged'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/activesupport-7.0.4.1/lib/active_support/tagged_logging.rb:99:in `block in tagged'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/railties-7.0.4.1/lib/rails/rack/logger.rb:25:in `block in call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/railties-7.0.4.1/lib/rails/rack/logger.rb:40:in `call_app'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/action_dispatch/middleware/show_exceptions.rb:26:in `call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/web-console-4.2.0/lib/web_console/middleware.rb:17:in `call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/web-console-4.2.0/lib/web_console/middleware.rb:17:in `catch'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/web-console-4.2.0/lib/web_console/middleware.rb:28:in `block in call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/web-console-4.2.0/lib/web_console/middleware.rb:132:in `call_app'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/action_dispatch/middleware/debug_exceptions.rb:28:in `call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/action_dispatch/middleware/actionable_exceptions.rb:17:in `call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/action_dispatch/middleware/executor.rb:14:in `call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/action_dispatch/middleware/callbacks.rb:26:in `call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/activesupport-7.0.4.1/lib/active_support/callbacks.rb:99:in `run_callbacks'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/action_dispatch/middleware/callbacks.rb:27:in `block in call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/activerecord-7.0.4.1/lib/active_record/migration.rb:603:in `call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/action_dispatch/middleware/cookies.rb:704:in `call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rack-2.2.6.2/lib/rack/session/abstract/id.rb:260:in `call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rack-2.2.6.2/lib/rack/session/abstract/id.rb:266:in `context'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/action_dispatch/http/content_security_policy.rb:36:in `call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/action_dispatch/http/permissions_policy.rb:38:in `call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rack-2.2.6.2/lib/rack/head.rb:12:in `call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rack-2.2.6.2/lib/rack/conditional_get.rb:40:in `call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rack-2.2.6.2/lib/rack/etag.rb:27:in `call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rack-2.2.6.2/lib/rack/tempfile_reaper.rb:15:in `call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/action_dispatch/routing/route_set.rb:852:in `call'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/action_dispatch/journey/router.rb:32:in `serve'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/action_dispatch/journey/router.rb:32:in `each'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/action_dispatch/journey/router.rb:50:in `block in serve'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/action_dispatch/routing/route_set.rb:32:in `serve'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/action_dispatch/routing/route_set.rb:49:in `dispatch'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/action_controller/metal.rb:251:in `dispatch'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/action_controller/metal.rb:188:in `dispatch'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionview-7.0.4.1/lib/action_view/rendering.rb:39:in `process'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/abstract_controller/base.rb:151:in `process'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/activerecord-7.0.4.1/lib/active_record/railties/controller_runtime.rb:27:in `process_action'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/action_controller/metal/params_wrapper.rb:259:in `process_action'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/action_controller/metal/instrumentation.rb:66:in `process_action'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/activesupport-7.0.4.1/lib/active_support/notifications.rb:206:in `instrument'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/activesupport-7.0.4.1/lib/active_support/notifications/instrumenter.rb:24:in `instrument'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/activesupport-7.0.4.1/lib/active_support/notifications.rb:206:in `block in instrument'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/action_controller/metal/instrumentation.rb:67:in `block in process_action'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/action_controller/metal/rescue.rb:22:in `process_action'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/abstract_controller/callbacks.rb:233:in `process_action'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/activesupport-7.0.4.1/lib/active_support/callbacks.rb:138:in `run_callbacks'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/activesupport-7.0.4.1/lib/active_support/callbacks.rb:127:in `block in run_callbacks'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/activesupport-7.0.4.1/lib/active_support/callbacks.rb:127:in `instance_exec'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actiontext-7.0.4.1/lib/action_text/engine.rb:69:in `block (4 levels) in <class:Engine>'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actiontext-7.0.4.1/lib/action_text/rendering.rb:20:in `with_renderer'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/activesupport-7.0.4.1/lib/active_support/callbacks.rb:118:in `block in run_callbacks'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/abstract_controller/callbacks.rb:234:in `block in process_action'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/action_controller/metal/rendering.rb:53:in `process_action'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/abstract_controller/base.rb:215:in `process_action'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/actionpack-7.0.4.1/lib/action_controller/metal/basic_implicit_render.rb:6:in `send_action'
/Users/dennis/RubymineProjects/csv_converter/app/controllers/csv_upload_controller.rb:44:in `edit'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/smarter_csv-1.7.4/lib/smarter_csv.rb:48:in `process'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/smarter_csv-1.7.4/lib/smarter_csv.rb:443:in `process_headers'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/smarter_csv-1.7.4/lib/smarter_csv.rb:258:in `parse'
/Users/dennis/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/smarter_csv-1.7.4/lib/smarter_csv.rb:258:in `parse_csv_line_c'
```

**Comment by tilo** [comment]
Thank you for pointing that out, @Shweit!

**Comment by tilo** [comment]
@Shweit this will be fixed in version 1.8.1
